### PR TITLE
MailQueueHandler: Catch provider errors

### PR DESCRIPTION
**Pull request by marcelklehr**

### CODE REVIEW

File: lib/MailQueueHandler.php [hunk 1]
```diff
@@ -434,7 +434,7 @@ protected function parseEvent($lang, IEvent $event) {
 		foreach ($this->activityManager->getProviders() as $provider) {
 			try {
 				$event = $provider->parse($lang, $event);
-			} catch (\InvalidArgumentException $e) {
+			} catch (\Throwable $e) {
```
Comment (artonge) [inline lib/MailQueueHandler.php:437]: Should we log the exception?
Can you give some explanation for the change?

**Review by nickvergessen — CHANGES_REQUESTED** [review]
At least without logging this is a very bad idea. It makes it impossible to find future errors by any app in their activity handling.

File: lib/MailQueueHandler.php (hunk 1 above)
Comment (nickvergessen) [reply lib/MailQueueHandler.php:437]: https://github.com/nextcloud/deck/pull/5533 + https://github.com/nextcloud/deck/pull/5542

But yeah also not a fan of it and I think we should not do this.

Comment (marcelklehr) [reply lib/MailQueueHandler.php:437]: Added logging

Comment (marcelklehr) [reply lib/MailQueueHandler.php:437]: The reason for this change is that a single error in some app can block the sending of all activity emails.

**Review by artonge — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Comment by marcelklehr** [comment]
It seems like part of the activity API is to allow providers to throw \InvalidArgumentExceptions when they don't recognize an event, which are now flooding the logs, because they are part of \Throwable.

**Comment by marcelklehr** [comment]
opened a PR: https://github.com/nextcloud/activity/pull/1586

**Pull request merged**